### PR TITLE
chore: rename dind-aws to dind-provider-cli, and add other provider (ibm, azure)

### PR DESCRIPTION
**Pull request by saugereau**
@Mallear , maybe if you can do some test on azure ...

### CODE REVIEW

File: config.yml [hunk 1]
```diff
@@ -76,18 +76,21 @@ chrome:
...
+dind-provider-cli:
   "1":
     build_args:
       CI_HELPER_VERSION: *CI_HELPER_VERSION
       DOCKER_COMPOSE_VERSION: 1.25.0
       GLIBC_VERSION: 2.27-r0
       PIP_VERSION: 18.1
       PIPENV_VERSION: 2018.10.13
+      IBM_CLI_VERSION: 0.22.0
+      AZURE_CLI_VERSION: 2.1.0
```
Comment (Mallear) [inline config.yml]: Latest azure cli version is 2.2.0

File: config.yml [hunk 2]
```diff
@@ -76,18 +76,21 @@ chrome:
...
     test_config:
       cmd:
-        - *ci_helper_test
         - aws --version
+        - az version
+        - ibmcloud version
```
Comment (Mallear) [inline config.yml]: Can you had a test on `ibmcloud cr` to check that CR plugin is well installed

File: config.yml (hunk 1 above)
Comment (saugereau) [reply config.yml]: Fixed

File: config.yml (hunk 2 above)
Comment (saugereau) [reply config.yml]: Fixed

File: .travis.yml [hunk 3]
```diff
@@ -18,7 +18,7 @@ matrix:
     - env: IMAGE=awslnx-systemd VERSION=2
     - env: IMAGE=azure VERSION=2
     - env: IMAGE=chrome VERSION=1
-    - env: IMAGE=dind-aws VERSION=1
+    - env: IMAGE=dind-provider-cli VERSION=1
```
Comment (rande) [inline .travis.yml]: Can you just renamed the image to `dind` ?

File: dind-provider-cli/Dockerfile [hunk 4]
```diff
@@ -1,24 +1,33 @@
...
 RUN echo "Install AWS" && \
-    apk add -q --no-cache bash build-base ca-certificates curl gettext git libffi-dev linux-headers make musl-dev openldap-dev openssh-client python py-pip rsync tzdata && \
-    pip install pip==${PYTHON_PIP_VERSION} && \
+    apk add -q --no-cache bash build-base ca-certificates curl gettext git libffi-dev linux-headers musl-dev openldap-dev openssh-client python3-dev gcc libffi-dev libressl-dev make rsync tzdata groff && \
+    python3 -m ensurepip && \
+    pip3 install --no-cache --upgrade pip setuptools wheel && \
     pip -q install pipenv==${PIPENV_VERSION} awscli boto3 PyYAML && \
     echo "Done install AWS" && \
 
-    echo "Install CI Helper" && \
```
Comment (rande) [inline dind-provider-cli/Dockerfile]: Please do not remove the CI Helper. We still need it.

File: .travis.yml (hunk 3 above)
Comment (saugereau) [reply .travis.yml]: Fixed

File: dind-provider-cli/Dockerfile (hunk 4 above)
Comment (saugereau) [reply dind-provider-cli/Dockerfile]: Fixed

File: CHANGELOG.md [hunk 5]
```diff
@@ -9,6 +9,7 @@ Versions
 * Move ibm image to Golang base image and build Null provider
 * Upgrade ibm provider to 1.2.4
 * Upgrade eksctl to 0.14.0
+* Rename dind-aws to dind-provider-cli and add all provider cli (aws, ibm, azure)
```
Comment (mremi) [inline CHANGELOG.md]:
```suggestion
* Rename dind-aws to dind and add all provider cli (aws, ibm, azure)
```

**Pull request merged**